ckanext/validation/tests/fixtures.py

```diff
@@ -19,3 +31,25 @@ def org():
 @pytest.fixture
 def dataset(org):
     return factories.Dataset(private=True, owner_org=org['id'])
+
+
+@pytest.fixture
+def mocked_responses():
+    solr_url = tk.config.get('solr_url', 'http://127.0.0.1:8983/solr/ckan')
+
+    with responses.RequestsMock() as rsps:
+        rsps.add_passthru(solr_url)
+        yield rsps
+
+
+class ResourceFactory(factories.Resource):
```

Comment:
Not included in 2.8, 2.9.x, only 2.10.x/master. It has not been backported and the dev requirements for this in on the on the main ckan-dev container. (i.e. you will need to include the correct py2 or py3 version to be backwards compatible.)

https://github.com/ckan/ckan/commit/6155426914ca8efa6562a0220fffc393a1174dc9#